src/commands/tail-cloudwatch-logs.js

```diff
@@ -0,0 +1,163 @@
+const _ = require("lodash");
+const { getAWSSDK } = require("../lib/aws");
+const { Command, flags } = require("@oclif/command");
+const { checkVersion } = require("../lib/version-check");
+const Promise = require("bluebird");
+const moment = require("moment");
+const inquirer = require("inquirer");
+
+class TailCloudwatchLogsCommand extends Command {
+	async run() {
+		const { flags } = this.parse(TailCloudwatchLogsCommand);
+		const { namePrefix, region, profile, filterPattern, interval } = flags;
+
+		global.region = region;
+		global.profile = profile;
+		global.filterPattern = filterPattern;
+		global.interval = interval || 1000;
+
+		checkVersion();
+
+		const AWS = getAWSSDK();
+		this.Logs = new AWS.CloudWatchLogs();
+
+		this.log(
+			`looking for CloudWatch log groups with prefix [${namePrefix}] in [${region}]`
+		);
+		const logGroupName = await this.getLogGroupName(namePrefix);
+
+		this.log(`looking for log streams for [${logGroupName}]...`);
+		const logStreamNames = await this.getLogStreamNames(logGroupName);
+
+		this.log(`polling CLoudWatch log group [${logGroupName}]...`);
+		this.log("press <any key> to stop");
+		await this.pollLogGroup(logGroupName, logStreamNames);
+
+		this.exit(0);
+	}
+
+	async getLogGroupName(namePrefix) {
+		const resp = await this.Logs.describeLogGroups({
+			logGroupNamePrefix: namePrefix
+		}).promise();
+
+		if (resp.logGroups.length === 1) {
+			this.log("only 1 match found");
+			return resp.logGroups[0].logGroupName;
+		} else if (_.isEmpty(resp.logGroups)) {
+			this.log(
+				"no matching log groups, please double check the prefix and region and try again"
+			);
+			this.exit(1);
+		} else if (resp.nextToken) {
+			this.log(
+				"too many log groups with matching prefix, please provide the full log group name and try again"
+			);
+			this.exit(1);
+		} else {
+			const logGroupChoices = resp.logGroups.map(x => x.logGroupName);
+			const { logGroupName } = await inquirer.prompt([
+				{
+					type: "list",
+					name: "logGroupName",
+					message: "Which log group are you looking for?",
+					choices: logGroupChoices
+				}
+			]);
+
+			return logGroupName;
+		}
+	}
+
+	async getLogStreamNames(logGroupName) {
+		const resp = await this.Logs.describeLogStreams({
+			logGroupName,
+			descending: true,
+			limit: 50,
+			orderBy: "LastEventTime"
+		}).promise();
+
+		if (_.isEmpty(resp.logStreams)) {
+			this.log("there are no log streams right now, please try again later");
+			this.exit(1);
+		} else {
+			return resp.logStreams.map(x => x.logStreamName);
+		}
+	}
+
+	async pollLogGroup(logGroupName, logStreamNames) {
+		let polling = true;
+		const readline = require("readline");
+		readline.emitKeypressEvents(process.stdin);
+		process.stdin.setRawMode(true);
+		const stdin = process.openStdin();
+		stdin.once("keypress", () => {
+			polling = false;
+			console.log("stopping...");
```

Comment:
```
polling = !polling
console.log(polling ? "continue..." : "stopping...")
```

Reply:
this bit of code is only run when you cancel the polling, so I don't think it's necessary